CMakeLists.txt

```diff
@@ -54,7 +33,31 @@ endif()
 target_include_directories(_isl PRIVATE ${ISL_INC_DIRS})
 
 if(USE_SHIPPED_ISL)
-  target_compile_definitions(_isl PRIVATE GIT_HEAD_ID="${ISL_GIT_HEAD_ID}")
+  add_library(isl STATIC ${ISL_SOURCES})
+  set_target_properties(isl PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+  target_link_libraries(_isl PRIVATE isl)
+
+  target_include_directories(isl PRIVATE ${ISL_INC_DIRS})
+
+  target_compile_definitions(isl PRIVATE GIT_HEAD_ID="${ISL_GIT_HEAD_ID}")
+  if(USE_IMATH_FOR_MP)
+    target_compile_definitions(isl PRIVATE USE_IMATH_FOR_MP=1)
+  endif()
+
+  if(USE_IMATH_SIO)
+    target_compile_definitions(isl PRIVATE USE_SMALL_INT_OPT=1)
+  endif()
+
+  if(USE_GMP_FOR_MP)
+    target_compile_definitions(isl PRIVATE USE_GMP_FOR_MP=1)
+  endif()
+
+  # Work around https://github.com/inducer/islpy/issues/120.
+  # See https://stackoverflow.com/questions/43554227/extern-inline-func-results-in-undefined-reference-error
+  # for some context.
+  set_source_files_properties(${ISL_SOURCES} PROPERTIES COMPILE_DEFINITIONS __OPTIMIZE_SIZE__)
```

Comment:
I think this line is only needed when `NOMINSIZE` is specified.

Reply:
Sure--but the isl sources are now being built without `-Os` as part of a separate build process (so probably with `-O2`).